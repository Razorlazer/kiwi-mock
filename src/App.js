import React from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import { ThemeProvider } from "@material-ui/styles";

import LandingPage from './pages/LandingPage';
import mainTheme from './utilities/theme/mainTheme';

import './App.css';

function App() {
  return (<>
    <ThemeProvider theme={mainTheme}>
    <CssBaseline/>
    <LandingPage />
    </ThemeProvider>
  </>);
}

export default App;
