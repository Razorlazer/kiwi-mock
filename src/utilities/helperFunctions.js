export const buildURLParams = (params = {})=>{
    const keys = Object.keys(params);

    const urlParams = [];
    for (let key = 0; key < keys.length; key++){
        if (params[keys[key]]){
            urlParams.push(`${keys[key]}=${params[keys[key]]}`)
        }
    }
  
    return urlParams.join('&');
};

export const createLocationSelectionList = ({ locations }, state) => {
    const mappedLocations = locations.filter(location => location.code !== state.destinationLocation && location.code!== state.departureLocation)
    .map(location => ({
        id: location.id,
        code: location.code,
        label: location.name,
        global_rank_dst: location.global_rank_dst,
        country: location.country,
        city: location.city
    }));

    return mappedLocations;
};

export const timeEpochUTCtoLocal = (dtime) => {
    var d = new Date(0); // The 0 there is the key, which sets the date to the epoch
    d.setUTCSeconds(dtime);
    return d.toLocaleString();
} 