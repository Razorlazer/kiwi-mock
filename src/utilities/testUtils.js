import configureStore from 'redux-mock-store';
import { flightsInitialState } from '../store/slices/flightsSlice';
import { locationsInitialState } from '../store/slices/locationsSlice';

export const fullSearchState = {
    locations: {
        ...locationsInitialState,
        departureLocation: 'PRG',
        destinationLocation: 'VIE',
    },
    flights: flightsInitialState
}
export const mockStore = (initialState) =>{ 
    return configureStore([])(initialState || {
        locations: locationsInitialState,
        flights: flightsInitialState
    });
}

