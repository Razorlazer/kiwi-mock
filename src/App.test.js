import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { mockStore } from './utilities/testUtils';
import App from './App';

test('renders learn react link', () => {
  const { asFragment } = render(
    <Provider store={mockStore()}>
      <App />
    </Provider>
  );

  expect(asFragment(<App />)).toMatchSnapshot();
});
