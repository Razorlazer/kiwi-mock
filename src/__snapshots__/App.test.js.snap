// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionStatic css-1cjrmu8-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="css-8psptw-MuiStack-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-iv7vbz-MuiGrid-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="location-input"
                  id="location-input-label"
                >
                  Departure city/airport
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="new-password"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="location-input"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Departure city/airport
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-iv7vbz-MuiGrid-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="location-input"
                  id="location-input-label"
                >
                  Destination city/airport
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="new-password"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="location-input"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Destination city/airport
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-1unik2p-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r4:"
                id=":r4:-label"
              >
                Departure date
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="Choose date"
                  aria-readonly="true"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r4:"
                  readonly=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Departure date
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-1unik2p-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r6:"
                id=":r6:-label"
              >
                Return date
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="Choose date"
                  aria-readonly="true"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r6:"
                  readonly=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Return date
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiGrid-root MuiGrid-container css-k9s8p3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-14xmr3p"
            >
              <div
                class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
                role="group"
              >
                <button
                  aria-pressed="true"
                  class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="M"
                >
                  Economy
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="W"
                >
                  Premium Economy
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="C"
                >
                  Business
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="F"
                >
                  First class
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-5 MuiGrid-grid-lg-5 css-urj5pv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  id="slider"
                >
                  Price up to:
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1v5z18m"
                >
                  <span
                    class="MuiSlider-colorPrimary MuiSlider-sizeSmall MuiSlider-root MuiSlider-marked css-xmer36-MuiSlider-root"
                  >
                    <span
                      class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
                    />
                    <span
                      class="MuiSlider-track css-1n40zqk-MuiSlider-track"
                      style="left: 0%; width: 25%;"
                    />
                    <span
                      class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
                      data-index="0"
                      style="left: 5%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
                      data-index="0"
                      style="left: 5%;"
                    >
                      100€
                    </span>
                    <span
                      class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
                      data-index="1"
                      style="left: 25%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
                      data-index="1"
                      style="left: 25%;"
                    >
                      500€
                    </span>
                    <span
                      class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
                      data-index="2"
                      style="left: 50%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
                      data-index="2"
                      style="left: 50%;"
                    >
                      1000€
                    </span>
                    <span
                      class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
                      data-index="3"
                      style="left: 100%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
                      data-index="3"
                      style="left: 100%;"
                    >
                      2000€
                    </span>
                    <span
                      class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeSmall MuiSlider-thumb css-14gf62f-MuiSlider-thumb"
                      data-focusvisible="false"
                      data-index="0"
                      style="left: 25%;"
                    >
                      <input
                        aria-label="slider"
                        aria-orientation="horizontal"
                        aria-valuemax="2000"
                        aria-valuemin="0"
                        aria-valuenow="500"
                        aria-valuetext="500€"
                        data-index="0"
                        max="2000"
                        min="0"
                        step="10"
                        style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                        type="range"
                        value="500"
                      />
                      <span
                        aria-hidden="true"
                        class="MuiSlider-valueLabel css-1xff9fg-MuiSlider-valueLabel"
                      >
                        <span
                          class="MuiSlider-valueLabelCircle"
                        >
                          <span
                            class="MuiSlider-valueLabelLabel"
                          >
                            500
                          </span>
                        </span>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1dfagy3-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-iv3xlb-MuiButtonBase-root-MuiButton-root"
              data-testid="search-button"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="css-16z22n8-MuiStack-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1pp3chc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
              >
                Welcome to minimalistic kiwi.com
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-ggu1h5-MuiTypography-root"
              >
                To find flights between cities, please use the search panel on top!
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1p5q5e5-MuiStack-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
